### PR TITLE
Added trait for capturing screenshots.

### PR DESCRIPTION
**Pull request by Chekote**

### CODE REVIEW

File: src/Behat/FlexibleMink/Context/ScreenShotContext.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+     *
+     * Note : If you wish to enable this in your tests, just add a method with an
+     *        `AfterStep` annotation to your Context as follows:
+     *
+     *        public function onFailedStep(AfterStepScope $scope) {
+     *          $this->takeScreenShotAfterFailedStep($scope);
+     *        }
+     *
+     * @param AfterStepScope $scope
+     */
```
Comment (crschoen) [inline src/Behat/FlexibleMink/Context/ScreenShotContext.php]: This documentation is duplicated on the abstract definition of `takeScreenShotAfterFailedStep`. I'm not sure that an `{@inheritdoc}` tag would work here, but perhaps an `@see`?

File: src/Behat/FlexibleMink/PseudoInterface/ScreenShotContextInterface.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    /**
+     * Captures a screenshot and saves it to the artifacts directory.
+     *
+     * @When  /^(?:I )?take a screenshot(?: named "(?P<name>(?:[^"]|\\")*)")$/
+     * @param string $name the name for the screenshot file (excluding path and extension)
+     */
+    abstract public function takeScreenShot($name = 'screenshot');
+
+    /**
+     * Captures screenshots of failed scenarios when running in Scrutinizer.
```
Comment (crschoen) [inline src/Behat/FlexibleMink/PseudoInterface/ScreenShotContextInterface.php]: Not just Scrutinizer.

File: src/Behat/FlexibleMink/Context/ScreenShotContext.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    // Implements
+    use ScreenShotContextInterface;
+
+    /**
+     * Captures a screenshot and saves it to the artifacts directory.
+     *
+     * @When   /^(?:I )?take a screenshot(?: named "(?P<name>(?:[^"]|\\")*)")$/
+     * @param  string $name the name for the screenshot file (excluding path and extension)
+     * @throws UnsupportedDriverActionException If the driver does not support getScreenshot().
+     */
```
Comment (crschoen) [inline src/Behat/FlexibleMink/Context/ScreenShotContext.php]: Please see my comment [here](https://github.com/Medology/FlexibleMink/pull/8/files#r59886605) about duplicate documentation.

File: src/Behat/FlexibleMink/PseudoInterface/MinkContextInterface.php [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+use Behat\Mink\Session;
+
+/**
+ * Pseudo interface for tracking the methods of the MinkContext.
+ *
+ * @see Behat\MinkExtension\Context\MinkContext.
+ */
+trait MinkContextInterface
+{
+
```
Comment (crschoen) [inline src/Behat/FlexibleMink/PseudoInterface/MinkContextInterface.php]: Please remove this blank line.

**Comment by crschoen** [comment]
First code review complete.

**Comment by Chekote** [comment]
@crschoen I guess I rushed the interface extraction a forgot to clean up those comments. Everything should be fixed now.

**Comment by crschoen** [comment]
Please add `composer.lock` to the `.gitignore` file.

**Pull request merged**